File: src/main/kotlin/mediumcodesamples/testdatabuilders/Person.java
package mediumcodesamples.testdatabuilders;

public class Person {

    private String name;
    private int age;
    private double weightInKg;
    private int heightInCm;
    private int energy;

    public Person(String name, int age, double weightInKg, int heightInCm, int energy) {
        this.name = name;
        this.age = age;
        this.weightInKg = weightInKg;
        this.heightInCm = heightInCm;
        this.energy = energy;
    }

    public int getEnergy() {
        return energy;
    }

    public void sleep() {
        this.energy = 100;
    }
}

File: src/main/kotlin/com/albertolopez/mediumcodesamples/testdatabuilders/Person.kt
package com.albertolopez.mediumcodesamples.testdatabuilders;

import java.math.RoundingMode

class Person(
    val name: String,
    val age: Int,
    val weightInKg: Double,
    val heightInM: Double,
    var calorieIntake: Int,
    var address: Address
) {

    init {
       if (age < 0) throw InvalidAgeException()
    }

    fun bmi(): Double {
        val weightInKgBD = weightInKg.toBigDecimal()
            .setScale(2, RoundingMode.HALF_EVEN)
        val heightInMSquaredBD = heightInM.toBigDecimal().pow(2)
        return weightInKgBD.div(heightInMSquaredBD).toDouble()
    }

    fun eatFood(calories: Int) {
        calorieIntake += calories
    }
}
